lib/client/db/dbcmd/error.go

```diff
@@ -44,6 +44,16 @@ func ConvertCommandError(cmd *exec.Cmd, err error, peakStderr string) error {
 				redisBin,
 			)
 		}
+	case mysqlBin, mariadbBin:
+		// Connection failure.
+		if strings.HasPrefix(peakStderr, "ERROR 1105 (HY000):") {
+			return trace.AccessDenied("'%s' could not connect to the database.", cmd.Path)
+		}
+	case postgresBin, cockroachBin:
+		// Connection failure.
+		if strings.HasPrefix(peakStderr, "psql: error: connection to server") {
```

Comment:
If a remove postgres server is unavailable for the db agent perspective the psql always returns the `psql: error: connection to server ` prefix: 
```
psql: error: connection to server at "teleport.example.com" (127.0.0.1), port 1234 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
```
So if those connectivity cases we will return AccessDenied error.

Reply:
Should we return `ConnectionProblem` for that  `connection to server` case? I can see the expired certificate error being considered a connection problem.